### ISSUE TITLE
auth: dnsupdate: allow-dnsupdate-from ALLOW-DNSUPDATE-FROM differ in behavior

### ISSUE THREAD

**Issue by jpmens** [PowerDNS/pdns#5855]
- Program: Authoritative
 - Issue type: Bug report

### Short description

`allow-dnsupdate-from` (from configuration) and `ALLOW-DNSUPDATE-FROM` (from metadata) differ in behavior.

### Environment
 - Operating system: CentOS
 - Software version: auth master `0.0.1909g0e9a0d5`
 - Software source: PowerDNS repository

#### pdns.conf

```
dnsupdate=yes
# allow-dnsupdate-from=10.1.1.1/32
```

#### domains

```
sqlite> select * from domains;
2|example.com|||NATIVE||
```

#### records

```
sqlite> select * from records;
1|2|example.com|NS|jp.vm.local|3600|0||0||1
6|2|www.example.com|A|1.1.1.2|10|0||0|www|1
8|2|www.example.com|A|1.1.1.1|60|0||0|www|1
9|2|example.com|SOA|jp.vm.local. jp.mens.de. 2017102624 10800 3600 604800 3600|3600|0||0||1
```

#### domainmetadata

```
sqlite> select * from domainmetadata;
3|2|ALLOW-DNSUPDATE-FROM|10.0.12.1/32
4|2|TSIG-ALLOW-DNSUPDATE|jpm-key
```

#### tsigkeys

```
sqlite> select * from tsigkeys;
1|jpm-key|hmac-sha256|oi0NEmZkV4qvRMFVrNyNHjQnwwnw/Tp1WFiVqq/4VVktRlF1amrwevx+hnuhvX9vSZ9WQlU9KOoJ0Rq4B4zVMg==
```

#### update.sh

```
nsupdate -y 'hmac-sha256:jpm-key:oi0NEmZkV4qvRMFVrNyNHjQnwwnw/Tp1WFiVqq/4VVktRlF1amrwevx+hnuhvX9vSZ9WQlU9KOoJ0Rq4B4zVMg==' <<EOF
server 127.0.0.1
zone example.com.
update add www.example.com 60 A 1.1.1.1
send
EOF
```

1. Update from 127.0.0.1 works although it should **not** because it's being restricted by `ALLOW-DNSUPDATE-FROM|10.0.12.1/32`.
2. Enabling (i.e. removing comment symbol on) `allow-dnsupdate-from=10.1.1.1/32` above, makes update from 127.0.0.1 fail which is correct as updates are then allowed from 10.1.1.1 only (together with the TSIG key)

This means, to me, that `allow-dnsupdate-from` in config and `ALLOW-DNSUPDATE-FROM` differ in behavior.

### Expected behaviour

I expect both types of configuration to behave similarly.

### Usecase

The use case is limiting updates to IP ranges together with TSIG keys.

**Comment by jpmens** [comment]
According to the documentation: 

> The option `[allow-dnsupdate-from]` can be left empty to disallow everything, this then should be used in combination with the ALLOW-DNSUPDATE-FROM domain metadata setting per zone.

I find this a bit confusing, but it appears to work. If this is done on purpose it needs more clarity in the documentation.

**Comment by jpmens** [comment]
I've added [documentation to settings](https://github.com/PowerDNS/pdns/pull/5857). I think with that, this can be closed.

**Comment by mind04** [comment]
Both values are honoured. Ranges in 'allow-dnsupdate-from' OR 'ALLOW_DNSUPDATE_FROM' are allowed to update.

**Comment by pieterlexis** [comment]
closed in #5857
